app/scripts/services/validateRegistrant.js

```diff
@@ -1,23 +1,23 @@
 'use strict';
 
 angular.module('confRegistrationWebApp')
-  .service('validateRegistrant', function validateRegistrant() {
+  .service('validateRegistrant', function validateRegistrant(util) {
 
-    var blockVisibleRuleCheck = function(block, registrant){
+    var blockVisibleRuleCheck = function (block, registrant) {
       var returnValue = true;
       var answers = registrant.answers;
-      angular.forEach(block.rules, function(rule){
-        var answer = _.find(answers, {blockId: rule.parentBlockId});
-        if(angular.isUndefined(answer) || answer.value === ''){
+      angular.forEach(block.rules, function (rule) {
+        var answer = _.find(answers, { blockId: rule.parentBlockId });
+        if (angular.isUndefined(answer) || answer.value === '') {
           returnValue = false;
```

Comment:
@OzzieOrca The _find function is inside the forEach loop even in the master branch. This is required, as we have to validate all the rules against all the answers above the question, for which the rule is to be applied.